server/src/main/java/org/elasticsearch/action/fieldcaps/FieldCapabilitiesIndexResponse.java

```diff
@@ -25,18 +26,28 @@
     private final String indexName;
     private final Map<String, IndexFieldCapabilities> responseMap;
     private final boolean canMatch;
+    private final Exception exception;
```

Comment:
I don't think the exception should be part of the index response. It's already reported in the coordinator node so we don't need this hybrid approach.  `FieldCapabilitiesIndexResponse` should represent a successful response.